aten/src/TH/THRandom.h

```diff
@@ -71,6 +71,11 @@ TH_API double THRandom_exponential(THGenerator *_generator, double lambda);
 */
 TH_API double THRandom_standard_gamma(THGenerator *_generator, double alpha);
 
+/** Computes a reparameterized gradient of a sample from a standard Gamma
+    distribution wrt the shape parameter alpha.
+*/
+TH_API double THRandom_standard_gamma_grad(double x, double alpha);
```

Comment:
I think it would be cleaner to just put the body of this function in the apply macro in THTensorRandom. All other functions in this file do actual sampling, while this is deterministic, and unlikely to be used outside of this single method (correct me if I’m wrong).

Reply:
Thanks for the suggestion. I've gone one step further and moved this to `THTensorMath.c` so it can use `THTensor(digamma_one)` internally once https://github.com/pytorch/pytorch/pull/3955 is merged. This seems like a better location than `THTensorRandom` because as you observe there is no random generation going on: this is merely a useful special function.